File: client/src/Types/member.ts
export type Member = {
  id: string;
  email: string;
  gender: string;
  dateOfBirth: string;
  displayName: string;
  created: string;
  lastActive: string;
  description?: string;
  city: string;
  country: string;
  imageUrl: string;
}
export type Photo = {
    id:number
    url:string
    publicId?: string
    memberId:string
}
export type EditableMember = {
  displayName: string;
  description?: string;
  city: string;
  country: string;

}
export class MemberParams{
  gender?:string;
  minAge=18;
  maxAge=100;
  pageNumber = 1 ;
  pageSize = 10;
  orderBy ='lastActive';
}

File: API/Entities/Member.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Member
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public DateOnly DateOfBirth { get; set; }

    public string? ImageUrl { get; set; }

    public required string DisplayName { get; set; }
    public DateTime Created { get; set; } = DateTime.UtcNow;
    public DateTime LastActive { get; set; } = DateTime.UtcNow;

    public required string Gender { get; set; }

    public string? Description { get; set; }

    public required string City { get; set; }

    public required string Country { get; set; }

    //Navigation Property
    [ForeignKey(nameof(Id))]
    [JsonIgnore]
    public AppUser User { get; set; } = null!;

    //nav property
    [JsonIgnore]// to ingore it while receving the response
    public List<Photo> Photos { get; set; } = [];

    [JsonIgnore]
    public List<MemberLike> LikeByMembers { get; set; } = [];
    
    [JsonIgnore]

    public List<MemberLike> LikedMembers { get; set; } = [];

}
